File: ip-sender.js
var http = require("http");
var salt = "=4-{.@hE]!EBk<o";

var optionsRead = {
    host: 'www.curlmyip.com',
    port: 80,
};

var get = http.request(optionsRead, function(res) {
    res.setEncoding('utf8');
    res.on('data', function (ip) {
        var optionsSend = {
            host: '127.0.0.1',
            port: 3000,
            method: 'POST',
            path: '/',
            'Content-Type': 'application/x-www-form-urlencoded'
        };

        var send = http.request(optionsSend, function(res) {
            res.setEncoding('utf8');
            res.on('data', function (chunk) {
                console.log(chunk);
            });
        });

        var postData = JSON.stringify({salt: salt, ip: ip});
        send.end(postData);
    });
});

get.end();

File: ip-sender.py
#!/bin/python
import http.client, urllib.parse
# get IP
conn = http.client.HTTPConnection("curlmyip.com")
conn.request("GET", "/")
r = conn.getresponse()
ip = r.read().rstrip()

# close conn
conn.close()

# post IP
postConn = http.client.HTTPConnection("dyn-ip.herokuapp.com")
params = urllib.parse.urlencode({'ip': ip})
headers = {"Content-type": "application/x-www-form-urlencoded", "Accept": "text/plain"}
postConn.request("POST", "", params, headers)
response = postConn.getresponse()
data = response.read()
print(data)

# close conn
postConn.close()
